crates/sui-indexer/src/apis/extended_api_v2.rs

```diff
@@ -54,6 +54,44 @@ impl ExtendedApiServer for ExtendedApiV2 {
         })
     }
 
+    async fn get_epoch_metrics(
+        &self,
+        cursor: Option<BigInt<u64>>,
+        limit: Option<usize>,
+        descending_order: Option<bool>,
+    ) -> RpcResult<EpochMetricsPage> {
+        let limit = validate_limit(limit, QUERY_MAX_RESULT_LIMIT_CHECKPOINTS)?;
+        let epochs = self
+            .inner
+            .spawn_blocking(move |this| {
+                this.get_epochs(
+                    cursor.map(|x| *x),
+                    limit + 1,
+                    descending_order.unwrap_or(false),
+                )
+            })
+            .await?;
+
+        let epoch_metrics = epochs
+            .into_iter()
+            .map(|e| EpochMetrics {
+                epoch: e.epoch,
+                epoch_total_transactions: e.epoch_total_transactions,
+                first_checkpoint_id: e.first_checkpoint_id,
+                epoch_start_timestamp: e.epoch_start_timestamp,
+                end_of_epoch_info: e.end_of_epoch_info,
+            })
+            .collect::<Vec<_>>();
+
+        let has_next_page = epoch_metrics.len() > limit;
+        let next_cursor = epoch_metrics.last().map(|e| e.epoch);
+        Ok(Page {
+            data: epoch_metrics,
```

Comment:
nit: do we not truncate the result set i.e. `epoch_metrics` to size `limit` (it seems like it could be of length `limit + 1` based on the params passed to `this.get_epochs` above?

Reply:
nice catch, thanks!